Assignment 2.Rmd

```diff
@@ -80,15 +94,72 @@ plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender)
 
 plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.width=EDGE$count)
-````
+```
 
 ## Part II
 
-In Part II your task is to [look up](http://igraph.org/r/) in the igraph documentation and create a graph that sizes the student vertices in terms of the number of disciplinary actions they have recieved, and the teachers in terms of the number of disciplinary actions they have given out.
+# In Part II your task is to [look up](http://igraph.org/r/) in the igraph documentation and create a graph that sizes the student vertices in terms of the number of disciplinary actions they have recieved, and the teachers in terms of the number of disciplinary actions they have given out.
+```{r}
+g <- graph.data.frame(EDGE, directed=TRUE, vertices=VERTEX)
+EDGES.st <- EDGE %>% group_by(to) %>% summarise(sum(count))
+names(EDGES.st) <- c("id", "count")
+EDGES.tc <- EDGE %>% group_by(from) %>% summarise(sum(count))
+names(EDGES.tc) <- c("id", "count")
+total <- bind_rows(EDGES.st,EDGES.tc)
+total
+VERTEX2 <- full_join(VERTEX, total, by = "id")
+g2 <- graph.data.frame(total, directed = TRUE, vertices = VERTEX2)
+plot(g2, layout = layout.fruchterman.reingold, vertex.color = VERTEX2$gender,vertex.size = VERTEX2$count+10, arrow.size=0.3)
+```
+
 
 ## Part III
 
 Now practice with data from our class. Please create a **person-network** with the data set hudk4050-classes.csv. To create this network you will need to create a person-class matrix using the tidyr functions and then create a person-person matrix using `t()`. You will then need to plot a matrix rather than a data frame using igraph.
 
 Once you have done this, also [look up](http://igraph.org/r/) how to generate the following network metrics: betweeness centrality and dregree. **Who is the most central person in the network?**
+```{r}
+# install.packages("tidyr")
+# install.packages("dplyr")
+library(tidyr)
+library(dplyr)
+
+# load file
+data1 <- read.csv("hudk4050-classes.csv", header = TRUE)
+# clean spaces
+data1<-lapply(data1, gsub, pattern=" ", replacement="")
+# unite names into one column
+data1<-as.data.frame(data1)
+data1<-unite(data1, StudentName, "First.Name", "Last.Name", sep=" ")
+# gather & select required data
+data2 <- data1 %>% gather (key = classnum, value = class, Class.1:Class.6)  %>% select (c(-classnum))
+
+data2 <- filter(data2, class >0)
+data2 <- filter(data2, class != "HUDK4050")
+data2$count<-1
+data2<-unique(data2)
+data3 <- data2 %>% spread(class, count) 
+
+# create matrix
+row.names(data3)<-data3$StudentName
+data3$StudentName<-NULL
+data3<-as.matrix(data3)
+data3 = ifelse(is.na(data3), 0, 1)
+data4<-data3 %*% t(data3)
+diag(data4)<-NA
+# graph
+g3<-graph.adjacency(data4, mode="undirected")
+plot(g3, layout=layout.fruchterman.reingold, vertex.label.size=0.5)
+
+betweenness(g3)
+degree(g3)
+which(betweenness(g3)==max(betweenness(g3)))   
+which(degree(g3)==max(degree(g3)))   
+# betweenness: Yujun Zhang 
+# degree: Lintong Li 
+```
+```
```

Comment:
Good job! Clear data wrangling and plotting. You can add attributes to show different numbers of common classes shared.